### ISSUE TITLE
Mist browser doesn't redirect as expected

### ISSUE THREAD

**Issue by morelazers** [ethereum/mist#1560]
First off - love the app, awesome work.

OSX Sierra
Mist 0.8.8

I'm using expressjs to build a dapp, and at certain points in the application, I'm using JavaScript to submit POST forms.

Within the request handler, there is code which redirects the browser back to the same page the POST request came from, but Mist isn't 'reloading' the page as it should.

I know this might be bad form, but on every other browser (that I've tried) it's not an issue, so if it's a design decision is there a reason specified?

I've tested by changing the URL that it's redirecting to, and it works great, issue only occurs when I'm trying to redirect to the same place.

Thanks!

**Comment by frozeman** [comment]
Hm that might be a special case where our routing doenstn properly work.
If you can give me a simple repro index.html i will try to fix it.

**Comment by morelazers** [comment]
Thanks @frozeman, I'm actually struggling to reproduce. It's odd in that the request is going through, but the response comes back as 'canceled' in the dev tools.
![image](https://cloud.githubusercontent.com/assets/2966123/22648223/8b26c60e-ec6d-11e6-961f-7d17154e2521.png)
I can confirm it's working in Chrome, are there any additional steps you think I should take to narrow down the problem?

**Comment by frozeman** [comment]
We handle the redirection in mist customised, so io would need a simple repo to be able to play around with it.

**Comment by morelazers** [comment]
@frozeman Yeah sorry - the source code is in the repo linked above - it should provide the same behaviour

https://github.com/morelazers/mist-redirect

**Comment by luclu** [comment]
I might be missing something here, but the test repo seems to work for me using Mist `0.8.9`, Firefox and Chromium. Here is a quick gif (5mb) of what happens in Mist:

![kapture 2017-03-14 at 19 07 18](https://cloud.githubusercontent.com/assets/10135872/23915583/d7a45b5c-08e9-11e7-81f3-0b58c010b1c3.gif)

**Comment by morelazers** [comment]
Hi @luclu, thanks for taking a look. My example wasn't clear enough, apologies. I've pushed an extra line or two to the repo which should make it clearer.

**Comment by luclu** [comment]
Thank you @morelazers, the change makes it clear. (Mist vs. Chromium):

![kapture 2017-03-14 at 22 04 54](https://cloud.githubusercontent.com/assets/10135872/23922382/6b7b25d2-0902-11e7-9741-7f5c7aa6bea2.gif)

@frozeman for your convenience: ;-)

```javascript
<script>
    $('form').on('submit', function() {
    $("#message").text('This should disappear')
	$(this).unbind('submit').submit();
    })
</script>
```
